internal/selector/selector.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package selector
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// MatchesSelector returns true if the selector set matches the values in the labels.
+func MatchesLabelSelector(selector metav1.LabelSelector, matchLabels map[string]string) bool {
```

Comment:
This feels like a little bit of a stutter, although it's technically correct. What do you think about `selector.MatchesLabels` instead?

Reply:
I had it like that first time in coding, but the label selector can be a set of labels or expressions so that's why I moved it back to have "selector" in the name.

Reply:
What about calling the package `match` and this function `LabelSelector`? Or possibly `compare` or `comparer`? Then we have a place to put other comparison operators, and we're not stuttering.